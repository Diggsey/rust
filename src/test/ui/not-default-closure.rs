// Check that closures do not implement `Default` if their environment is not empty.

fn test_default<F: Default + Fn()>(_: F) {
    let f = F::default();
    f();
}

fn main() {
    let a = "Bob";
    let hello = move || {
        println!("Hello {}", a);
    };

    test_default(hello); //~ ERROR the trait bound
}
