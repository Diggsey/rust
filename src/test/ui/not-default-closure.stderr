error[E0277]: the trait bound `[closure@$DIR/not-default-closure.rs:10:17: 12:6]: Default` is not satisfied
  --> $DIR/not-default-closure.rs:14:5
   |
LL | fn test_default<F: Default + Fn()>(_: F) {
   |                    ------- required by this bound in `test_default`
...
LL |     let hello = move || {
   |                 ------- consider calling this closure
...
LL |     test_default(hello);
   |     ^^^^^^^^^^^^ the trait `Default` is not implemented for `[closure@$DIR/not-default-closure.rs:10:17: 12:6]`
   |
   = help: use parentheses to call the closure: `hello()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
